model/github/github_repository.rb

```diff
@@ -46,17 +46,24 @@ def admin_client
   end
 
   def destroy_blob_storage
-    admin_client.delete_bucket(bucket: bucket_name)
+    begin
+      admin_client.delete_bucket(bucket: bucket_name)
+    rescue Aws::S3::Errors::NoSuchBucket
+    end
+
     CloudflareClient.new(Config.github_cache_blob_storage_api_key).delete_token(access_key)
     update(access_key: nil, secret_key: nil)
   end
 
   def setup_blob_storage
     DB.transaction do
-      admin_client.create_bucket({
-        bucket: bucket_name,
-        create_bucket_configuration: {location_constraint: Config.github_cache_blob_storage_region}
-      })
+      begin
+        admin_client.create_bucket({
+          bucket: bucket_name,
+          create_bucket_configuration: {location_constraint: Config.github_cache_blob_storage_region}
+        })
+      rescue Aws::S3::Errors::BucketAlreadyOwnedByYou
```

Comment:
from gut feeling I would have expected an error like `Aws::S3::Types::BucketAlreadyExists` in this case. but `Aws::S3::Errors::BucketAlreadyOwnedByYou` seems correct.

Reply:
It was a shock for me as well. I considered adding both exceptions, but bucket names in Cloudflare R2 are not globally unique. They are unique only within an account, so Cloudflare R2 never throws an `Aws::S3::Types::BucketAlreadyExists` error. However, since bucket names in AWS S3 are globally unique, I believe the `Aws::S3::Types::BucketAlreadyExists` error is thrown when a bucket with the same name exists in any other account.